modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -920,7 +921,7 @@ public CPDefinition copyCPDefinition(
 		ServiceContext serviceContext =
 			ServiceContextThreadLocal.getServiceContext();
 
-		User user = _userLocalService.getUser(serviceContext.getUserId());
+		User user = _userLocalService.getUser(PrincipalThreadLocal.getUserId());
```

Comment:
@MahmoudAzzam please remove the 
`ServiceContext serviceContext =
			ServiceContextThreadLocal.getServiceContext();`

instruction at line 921.

Thanks!

Reply:
@andrea-ale-sbarra @luca-pellizzon This will break the code as it is used at line 1261 and 1274

Reply:
@MahmoudAzzam as per the ticket we need to remove ServiceContextThreadLocal.getServiceContext() from that class. So you can replace the serviceContext with the parameters needed in all the methods of the chain or you can pass through the serviceContext to copyCPDefinition instead of using the ThreadLocal.
cc @alessiorendina